stable/jenkins/templates/config.yaml

```
@@ -248,12 +264,46 @@ data:
 {{- range $key, $val := .Values.Master.InitScripts }}
   init{{ $key }}.groovy: |-
 {{ $val | indent 4 }}
+{{- end }}
+{{- if and (.Values.Master.JCasC.enabled) (.Values.Master.Sidecar.configAutoReload.enabled) }}
+  init-add-ssh-key-to-admin.groovy: |-
+    import jenkins.security.*
+    import hudson.model.User
+    import jenkins.security.ApiTokenProperty
+    import jenkins.model.Jenkins
+    User u = User.get("{{ .Values.Master.AdminUser | default "admin" }}")
+    ApiTokenProperty t = u.getProperty(ApiTokenProperty.class)
+    String sshKeyString = new File('/var/jenkins_home/key.pub').text
+    keys_param = new org.jenkinsci.main.modules.cli.auth.ssh.UserPropertyImpl(sshKeyString)
+    u.addProperty(keys_param)
+    def inst = Jenkins.getInstance()
+    def sshDesc = inst.getDescriptor("org.jenkinsci.main.modules.sshd.SSHD")
+    sshDesc.setPort({{ .Values.Master.Sidecar.configAutoReload.sshTcpPort | default 1044 }})
+    sshDesc.getActualPort()
+    sshDesc.save()
+{{- end }}
+# Only add config to this script if we aren't auto-reloading otherwise the pod will restart upon each config change:
+{{- if .Values.Master.JCasC.enabled }}
+{{- if not .Values.Master.Sidecar.configAutoReload.enabled }}
+{{- range $key, $val := .Values.Master.JCasC.ConfigScripts }}
+  {{ $key }}.yaml: |-
+{{ $val | indent 4 }}
```

```suggestion
{{ tpl $val $| indent 4 }}
```
This would allow templating for JCasC config snippets, such as for the welcome message to include the name of the instance:
`
    ConfigScripts:
      welcomemessage: |
        jenkins:
          systemMessage: "{{ template "jenkins.fullname" . }} configured automatically by Helm Chart & Jenkins Configuration as Code Plugin\n\n"
`